explorer/src/Pos/Explorer/Web/Api.hs

```diff
@@ -18,85 +19,104 @@ module Pos.Explorer.Web.Api
 
 import           Universum
 
+import           Control.Monad.Catch          (try, tryJust)
 import           Data.Proxy                   (Proxy (Proxy))
 
 import           Pos.Explorer.Web.ClientTypes (Byte, CAda, CAddress, CAddressesFilter,
                                                CAddressSummary, CBlockEntry, CBlockSummary,
                                                CGenesisAddressInfo, CGenesisSummary,
                                                CHash, CTxBrief, CTxEntry, CTxId,
                                                CTxSummary)
-import           Pos.Explorer.Web.Error       (ExplorerError)
+import           Pos.Explorer.Web.Error       (ExplorerError, _Internal)
 import           Pos.Types                    (EpochIndex)
+import           Pos.Util.Servant             (ModifiesApiRes (..), VerbMod)
 import           Servant.API                  ((:<|>), (:>), Capture, Get, JSON,
                                                QueryParam)
 
-
 type PageNumber = Integer
 
+-- | API result modification mode used here.
+data ExplorerVerbTag
+
+-- | Wrapper for Servants 'Verb' data type,
+-- which allows to catch exceptions thrown by Explorer's endpoints.
+type ExplorerVerb verb = VerbMod ExplorerVerbTag verb
+
+-- | Shortcut for common api result types.
+type ExRes verbMethod a = ExplorerVerb (verbMethod '[JSON] a)
+
+instance ModifiesApiRes ExplorerVerbTag where
+    type ApiModifiedRes ExplorerVerbTag a = Either ExplorerError a
+    modifyApiResult _ =
+        try . catchEndpointErrors . (either throwM pure =<<)
```

Comment:
I prefer to have a parameter here than to have this chain and wonder what's _really_ going on.

Reply:
Would do you mean under parameter here?

Reply:
```modifyApiResult _ = try . catchEndpointErrors . (either throwM pure =<<)```
is pointfree style of following:
```modifyApiResult proxy action = try . try $ (either throwM pure =<<) action```.

We can't set a parameter `proxy`, because it is not used at the right hand side of `modifyApiResult`. We will get a warning instead: 
```
cardano-sl/explorer/src/Pos/Explorer/Web/Api.hs:57:21: warning: [-Wunused-matches]
    Defined but not used: ‘proxy’ 
```

But we could use 
```modifyApiResult _ action = try . try $ (either throwM pure =<<) action``` 
to avoid to have a pointfree style. 

@ksaric Would the last version (not pointree) of `modifyApiResult` ^ fix your concerns?

Reply:
@sectore Yes, I presume we don't need `proxy` here.
It's better, but I still don't get why we have 2 `try`s here. And `pure bind` (`pure =<<`).
Why not `try (either throwM action)`? Isn't that sufficient? I still haven't compiled this locally, so I'm running blind here. Nonetheless, it looks kind of boilerplate-y.

Reply:
@ksaric Yeah, good to have these questions! Because it helps me to understand everything more and more, too :) 
```
try (either throwM pure action)
```
will throw
```
Couldn't match expected type ‘Either e0 (Either ExplorerError a)’
                  with actual type ‘IO (Either ServantErr a)’
    • In the third argument of ‘either’, namely ‘action’
      In the first argument of ‘try’, namely
        ‘(either throwM pure action)’
      In the expression: try (either throwM pure action)
...
``` 
Agree, the best thing is to play around to answer your questions by `ghc`.

Another questions for me was: Why the hell do we need to add `Proxy ExplorerVerbTag`? It's because type families are not injective (we are using type families of Servant) and `ghc` won't be able to know about `ExplorerVerbTag`. (see https://arow.info/blog/posts/2015-07-10-servant-intro.html#why-pass-layout-twice BTW: This article helps me a lot to understand what's really going on here).

Reply:
I will take a look at this `Either e0 (Either ExplorerError a)` thing tomorrow, thanks.

Reply:
Yep, two `try`s there because we want to return `Either ServantErr (Either ExplorerError a)`. First exception is servant-required, and second one is what we want to return.

Reply:
Btw, in wallet code we also cough _all_ other exceptions here and converted them to `ServantErr` or `WalletError`, because otherwise on unexpected error we just got `500 Internal error` without any explanations of what's exactly going on; maybe it also occurred to you to suffer from it )